aspnetcore/fundamentals/http-context/samples/6.x/HttpContextInBackgroundThread/EmailService.cs

```diff
@@ -1,31 +1,43 @@
 namespace HttpContextInBackgroundThread;
 
+public interface IEmailService
+{
+    Task SendEmail(string email, string? userAgent = null);
+}
+
 public class EmailService : IEmailService
 {
     private readonly IHttpContextAccessor _httpContextAccessor;
     private readonly ILogger<EmailService> _logger;
-    private const string? BackgroundCorrelationId = "background-correlation-id";
 
     public EmailService(IHttpContextAccessor httpContextAccessor, ILogger<EmailService> logger)
     {
         _httpContextAccessor = httpContextAccessor;
         _logger = logger;
     }
 
-    public void SendEmail(string email)
+    // The userAgent should come directly from the caller to the service if possible.
+    // An explicit parameter makes the API more usable outside of the request flow, is
+    // better for performance and is easier to reason about than relying on ambient state.
+    public async Task SendEmail(string email, string? userAgent = null)
     {
-        // Services should account for the possibility of HttpContext being null if not called from a request thread.
-        // e.g. When a service is called from a BackgroundWorker the HttpContext is null.
-        // The service should either throw or gracefully handle this.
+        var request = _httpContextAccessor.HttpContext?.Request;
+        string ? userAgentString = userAgent ?? request?.Headers["user-agent"].ToString();
+        if (string.IsNullOrEmpty(userAgentString))
+        {
+            userAgentString = "Unkown";
+        }
 
-        var correlationId = _httpContextAccessor.HttpContext?.Request.Headers["X-Correlation-Id"].ToString() ?? BackgroundCorrelationId;
-
-        _ = SendEmailCoreAsync(correlationId);
+        _ = SendEmailCoreAsync(userAgentString);
+        //await _taskQueue.QueueBackgroundWorkItemAsync(cancellationToken => 
+        //                           SendEmailCoreAsync(userAgentString, cancellationToken));
```

Comment:
@halter73 could we replace the sample with the new sample in this comment? https://github.com/dotnet/AspNetCore.Docs/pull/25839#issuecomment-1148114428

Reply:
I do like demonstrating how a service would really likely be used in the background while the application is running using a `BackgroundService` with proper error and shutdown handling rather than calling `app.Services.GetService()` once and invoking the service once before starting.

I do see how demonstrating this using a `DelegatingHandler` for `IHttpClientFactory` rather than a fake `IEmailService` is a bit more fleshed out. I don't really see the motivation for then using that delegating handler to log GitHub branches at an information level, but maybe it can be updated to have the request handler and background service just used `IHttpClientFactory` directly rather than using the `BranchesLoggerService`.